### DISCUSSION TITLE
Issue with Sending Stream after Reconnecting to Ant Media Server Android SDK

### DISCUSSION THREAD

**Discussion by KavyashreeBasavesh** [ant-media/Ant-Media-Server#5449] [FAQs]
We are encountering a challenge related to the Ant Media Server Android SDK while dealing with stream reconnection scenarios. I would like to provide you with a detailed overview of the issue we are facing.

We have two devices, Device A (using the Android SDK) and Device B (also using the Android SDK), participating in a room where they exchange audio and video streams. In our use case, if Device A experiences a loss of connectivity and is subsequently turned off, it disconnects from the room. However, the room remains active with Device B still present.

When Device A is powered back on and reconnects to the same room, it successfully receives streams from Device B. However, we've noticed that Device A is unable to send its own stream.

We are using the Ant Media Server Android SDK for Device A and the Android SDK for Device B.

To address this issue, we are seeking your insights into the following:

How does the Android SDK notify the client application about the reconnection state?
Once the reconnection is established, how does the SDK inform the client application about it?
Which delegate of the AntMediaClient/WebRTCClient should we rely on to manage the display of the "Reconnection..." status and the actions related to replaying/republishing streams?

**Comment by USAMAWIZARD** [comment]
Hi , @KavyashreeBasavesh is this problem reproducible on Sample conference call application or is it something related to your own development

**Reply by KavyashreeBasavesh** [reply]
Hi @USAMAWIZARD Yes, it is repoducible.

**Reply by USAMAWIZARD** [reply]
Can you pleas check if its correct reproduce scenario for reproducing this case with example pages.

1. Publish from Device A and B.
2. Disconnect Device B from network
3. reconnect device B to network
4. device A will receive stream from device B but device B will not receive device  A stream.

**Reply by KavyashreeBasavesh** [reply]
Yes it is correct. Sometimes it may reverse.

**Reply by USAMAWIZARD** [reply]
There has been some recent improvement for this let me check and get get back to you.

**Reply by USAMAWIZARD** [reply]
Hi @KavyashreeBasavesh ,
I have just tested the re connection on the latest master branch and its working fine for me , please let me know if it does not works for you.
